pkg/prometheus/resource_selector.go

```diff
@@ -829,7 +829,7 @@ func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context, listFn List
 			continue
 		}
 
-		if err = rs.validateAzureSDConfigs(ctx, sc); err != nil {
+		if err = rs.validateAzureSDConfigs(ctx, cpf.Version, sc); err != nil {
```

Comment:
rs already knows about the monitoringv1.PrometheusInterface object so no need to pass the version.